lib/webtrit_adapter_client.ex

```diff
@@ -230,6 +230,26 @@ defmodule WebtritAdapterClient do
     request(client, options)
   end
 
+  @spec submit_sip_registration_status(Tesla.Client.t(), String.t(), any(), String.t() | nil) :: result()
+  def submit_sip_registration_status(client, status, timestamp, reason \\ nil) do
+    body =
+      %{
+        status: status,
+        timestamp: timestamp
+      }
+      |> then(fn map ->
+        if reason, do: Map.put(map, :reason, reason), else: map
+      end)
+
+    options = [
+      method: :post,
+      url: "/user/sip-registration-status",
+      body: body
+    ]
+
+    request(client, options)
+  end
```

Comment:
As for me such type of endpoint is to specific. I propose to make it more general, so in future we could easily extend it with other statuses.
My proposition is to name this endpoint `/user/events` with following body structure:
```
%{
  timestamp: <required value>,
  group: <required value form predefined enum, for now with one value "sip">,
  type: <required value form predefined enum related to group, for now with two values "registered" or "unregistered">,
  data: <optional additional map data of event, for example for "sip" group it could be %{code: ..., reason: ...}>
}
```

And yes `unregistered` is more natural naming for event then `notregistered`, where `notregistered` is more like consequence of `unregistered` event.

Reply:
Alternatively, instead of using `/user/events`, let’s use `/user/core-events` to clearly define the purpose of the events.